app/Observers/SignupObserver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rogue\Observers;
+
+use Rogue\Models\Signup;
+use Rogue\Services\GraphQL;
+
+const USER_CLUB_ID_QUERY = '
+    query UserClubIdQuery($userId: String!) {
+        user(id: $userId) {
+            clubId
+        }
+    }
+';
+
+class SignupObserver
+{
+    /**
+     * Query to get the user's club_id.
+     *
+     * @param string $userId
+     * @return array
+     */
+    public function queryForUser($userId)
+    {
+        return app(GraphQL::class)->query(USER_CLUB_ID_QUERY, [
+            'userId' => $userId,
+        ]);
+    }
+
+    /**
+     * Handle the Signup "creating" event.
+     *
+     * @param  \Rogue\Models\Signup  $signup
+     * @return void
+     */
+    public function creating(Signup $signup)
+    {
+        if (!$signup->club_id) {
+            $data = $this->queryForUser($signup->northstar_id);
```

Comment:
I am a _little_ worried about the devopswords here, since it seems like we'd now effectively be hitting Northstar for any signup that's created (so a spike in traffic to Rogue means a spike in traffic to Northstar). This absolutely feels like the cleanest way to implement this though, so I think it's good to keep & see if it causes us trouble later.

I think this is another good example of where the user/activity domains are somewhat awkward to split... it makes me continue to think about the idea of merging these two applications. 🤔

Reply:
lol ok I'm going to put devopsword on a t-shirt!

Sounds good -- I'll try and keep a close eye on this particularly around the next broadcast. Do you think it's worth wrapping this in a feature flag to be safe so we can flip it off if chaos starts to abound?

> I think this is another good example of where the user/activity domains are somewhat awkward to split... it makes me continue to think about the idea of merging these two applications. 🤔

👀 👀

Reply:
Ohhh, yeah feature flagging this would be dajirt if it's not too much work!

Reply:
Wrapped it all up in a (poorly named? but eh) feature flag in  https://github.com/DoSomething/rogue/pull/1121/commits/9e706a9ec227ed11d39f6124c0886bb19a49236e.

Reply:
🙌